python/pyspark/sql/catalog.py

```diff
@@ -255,26 +255,67 @@ def registerFunction(self, name, f, returnType=StringType()):
         >>> _ = spark.udf.register("stringLengthInt", len, IntegerType())
         >>> spark.sql("SELECT stringLengthInt('test')").collect()
         [Row(stringLengthInt(test)=4)]
+        """
+
+        # This is to check whether the input function is a wrapped/native UserDefinedFunction
+        if hasattr(f, 'asNondeterministic'):
```

Comment:
It might be better to check `f` is (1) callable (2) not a UDF object

Reply:
Yea, but the wrapped UDF is also callable .. unfortunately :(.

I suggested this way for this reason:

in `udf.py`

```diff
+        wrapper._unwrapped = lambda: self
         return wrapper
```

and then

```
if hasattr(f, "_unwrapped"):
    f = f._unwrapped()
if isinstance(f, UserDefinedFunction):
    ...
else:
    ...
```

but it was no string opinion.

Reply:
The current way is simple. Any hole?

Reply:
Nope, to be clear I am fine with it as is.

Reply:
I am fine with it as well.